src/claims.py

```diff
@@ -1,35 +1,76 @@
-import json
 from dataclasses import dataclass, field
-from typing import Any, Dict
+from datetime import datetime, timedelta
+from typing import Dict
 
+from jose import jwt  # type: ignore # pylint: disable=import-error
 
+from src.base import BaseJCSerializable, KeyMapping
+from src.errors import EARValidationError
+from src.jwt_config import DEFAULT_ALGORITHM, DEFAULT_EXPIRATION_MINUTES
+from src.submod import Submod
+from src.verifier_id import VerifierID
+
+
+# https://datatracker.ietf.org/doc/draft-fv-rats-ear/
 @dataclass
-class EARClaims:
+class AttestationResult(BaseJCSerializable):
     profile: str
     issued_at: int
-    verifier_id: Dict[str, str] = field(default_factory=dict)
-    submods: Dict[str, Any] = field(default_factory=dict)
+    verifier_id: VerifierID
+    submods: Dict[str, Submod] = field(default_factory=dict)
 
-    def to_dict(self) -> Dict[str, Any]:
-        return {
-            "eat_profile": self.profile,
-            "iat": self.issued_at,
-            "ear.verifier-id": self.verifier_id,
-            "submods": self.submods,
-        }
+    # https://www.ietf.org/archive/id/draft-ietf-rats-eat-31.html#section-7.2.4
+    jc_map = {
+        "profile": KeyMapping(265, "profile"),
+        "issued_at": KeyMapping(6, "issued_at"),
+        "verifier_id": KeyMapping(1004, "verifier_id"),
```

Comment:
The string name should be `"ear.verifier-id"`. Please note that JSON field names are defined by [draft-fv-rats-ear](https://datatracker.ietf.org/doc/html/draft-fv-rats-ear#name-new-eat-claims) (see "JWT Claim Name" entry). The same goes for other claim names as well -- please make sure they align with the spec.